x-pack/legacy/plugins/maps/public/layers/joins/left_inner_join.test.js

```diff
@@ -67,4 +70,76 @@ describe('joinPropertiesToFeature', () => {
       iso2: 'CN',
     });
   });
+
+
+  it('Should coerce to string before joining', () => {
+
+    const leftJoin = new LeftInnerJoin({
+      leftField: 'zipcode',
+      right: rightSource
+    });
+
+    const feature = {
+      properties: {
+        zipcode: 40204
+      }
+    }
+    ;
+    const propertiesMap = new Map();
+    propertiesMap.set('40204', { [COUNT_PROPERTY_NAME]: 61 });
+
+    leftJoin.joinPropertiesToFeature(feature, propertiesMap, [{
+      propertyKey: COUNT_PROPERTY_NAME
+    }]);
+    expect(feature.properties).toEqual({
+      zipcode: 40204,
+      [COUNT_PROPERTY_NAME]: 61,
+    });
+  });
+
+  it('Should handle undefined values', () => {
+
+    const feature = {//this feature does not have the iso2 field
+      properties: {
+        zipcode: 40204
+      }
+    }
+    ;
+    const propertiesMap = new Map();
+    propertiesMap.set('40204', { [COUNT_PROPERTY_NAME]: 61 });
+
+    leftJoin.joinPropertiesToFeature(feature, propertiesMap, [{
+      propertyKey: COUNT_PROPERTY_NAME
+    }]);
+    expect(feature.properties).toEqual({
+      zipcode: 40204
+    });
+  });
+
+  it('Should handle falsy values', () => {
+
+    const leftJoin = new LeftInnerJoin({
+      leftField: 'code',
+      right: rightSource
+    });
+
+    const feature = {//this feature does not have the iso2 field
```

Comment:
comment "this feature does not have the iso2 field" is not relevant here and can be removed